src/test/scala/fi/oph/koski/api/OppijaValidationTaiteenPerusopetusSpec.scala

```diff
@@ -108,6 +109,222 @@ class OppijaValidationTaiteenPerusopetusSpec
+        putOpiskeluoikeus(
+          TPO.Opiskeluoikeus.hankintakoulutuksenaHyväksytystiSuoritettuLaajaOppimäärä,
+          henkilö = oppija,
+          headers = authHeaders(MockUsers.varsinaisSuomiPalvelukäyttäjä) ++ jsonContent
+        ) {
+          verifyResponseStatus(403, KoskiErrorCategory.forbidden.vainTaiteenPerusopetuksenJärjestäjä())
+        }
+      }
+
+      "väärän oppilaitoksen käyttäjä ei voi luoda hankintakoulutuksena järjestettävää opiskeluoikeutta" in {
+        putOpiskeluoikeus(
+          TPO.Opiskeluoikeus.hankintakoulutuksenaHyväksytystiSuoritettuLaajaOppimäärä,
+          henkilö = oppija,
+          headers = authHeaders(MockUsers.stadinAmmattiopistoPääkäyttäjä) ++ jsonContent
+        ) {
+          verifyResponseStatus(403, KoskiErrorCategory.forbidden.vainTaiteenPerusopetuksenJärjestäjä())
+        }
+      }
+    }
+
+    "Opiskeluoikeuden muokkaamisen oikeudet" - {
+      "pääkäyttäjä voi muokata hankintakoulutuksena järjestettävän opiskeluoikeuden" in {
+        putOpiskeluoikeus(
+          TPO.Opiskeluoikeus.hankintakoulutuksenaHyväksytystiSuoritettuLaajaOppimäärä.copy(
+            arvioituPäättymispäivä = Some(LocalDate.now().plusDays(1))
+          ),
+          henkilö = KoskiSpecificMockOppijat.taiteenPerusopetusHankintakoulutus,
+          headers = authHeaders(MockUsers.paakayttaja) ++ jsonContent
+        ) {
+          verifyResponseStatusOk()
+        }
+      }
+
+      "koulutustoimijan käyttäjä voi muokata hankintakoulutuksena järjestettävän opiskeluoikeuden" in {
+        putOpiskeluoikeus(
+          TPO.Opiskeluoikeus.hankintakoulutuksenaHyväksytystiSuoritettuLaajaOppimäärä.copy(
+            arvioituPäättymispäivä = Some(LocalDate.now().plusDays(2))
+          ),
+          henkilö = KoskiSpecificMockOppijat.taiteenPerusopetusHankintakoulutus,
+          headers = authHeaders(MockUsers.helsinkiTallentaja) ++ jsonContent
+        ) {
+          verifyResponseStatusOk()
+        }
+      }
+
+      "väärän koulutustoimijan käyttäjä ei voi muokata hankintakoulutuksena järjestettävää opiskeluoikeutta" in {
+        val resp = putOpiskeluoikeus(
+          TPO.Opiskeluoikeus.hankintakoulutuksenaHyväksytystiSuoritettuLaajaOppimäärä.copy(
+            arvioituPäättymispäivä = Some(LocalDate.now().plusDays(3))
+          ),
+          henkilö = KoskiSpecificMockOppijat.taiteenPerusopetusHankintakoulutus,
+          headers = authHeaders(MockUsers.tornioTallentaja) ++ jsonContent
+        ) {
+          verifyResponseStatus(200)
+          readPutOppijaResponse
+        }
+        resp.opiskeluoikeudet.size shouldBe 1
+        val oo = oppija(resp.henkilö.oid).opiskeluoikeudet.find(_.oid.contains(resp.opiskeluoikeudet.head.oid))
+        oo.head.koulutustoimija.head.oid == MockOrganisaatiot.tornionKaupunki
+      }
+
+      "hankintakoulutuksen oppilaitoksen käyttäjä ei voi muokata hankintakoulutuksena järjestettävää opiskeluoikeutta" in {
```

Comment:
Tämä testi pitäisi palauttaa 200, oppilaitoksen käyttäjän pitäisi pystyä muokkaamaan oman oppilaitoksen opiskeluoikeutta, vaikka koulutustoimija on joku muu kaupunki.

Reply:
Harkinnan jälkeen mielestäni näin pitääkin olla, että tavallinen oppilaitos-tallentaja ei voi muokata hankintakoulutuksena järjestettävää opiskeluoikeutta, koska ei myöskään saa lisätä uutta hankintakoulutuksena järjestettävää opiskeluoikeutta. Hankintakoulutuksena järjestettävän opiskeluoikeuden muokkamiseen annetaan erillinen käyttöoikeus oppilaitoksen tallentajalle.